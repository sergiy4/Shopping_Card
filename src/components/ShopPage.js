import { useRef, useState } from "react"
import { plantsData } from "../data/PlantsListData"
import { useSpring, useSprings, animated } from "@react-spring/web"
import '../styles/ShopPage.css'

import { Parallax, ParallaxLayer } from '@react-spring/parallax'

function ShopPage(){

    const [index, setIndex] = useState(null);
    const [viewBtn, setViewBtn] = useState(false);
    const [BuyBtn, setBuyBtn] = useState(false);
    const [on,setOn] = useState(true)
    // const triggerRef = useRef();
    // const dataRef = useIntersectionObserver(triggerRef,{
    //     freezeOnceVisible: true
    // })
    const springs = useSprings(
        plantsData.length,
        plantsData.map((item,i)=>(
            {   
                from:{
                  
                    boxShadow:'0px 0px 0px 0px rgb(0, 0, 0)',
                    top:'0%',
                    left:'0%',
                },
                to:{

                    boxShadow: i === index? '10px 10px 0px 0px rgb(0, 0, 0)':'0px 0px 0px 0px rgb(0, 0, 0)' ,
                   

                    top: i=== index? '-3%':'0%' ,
                    left: i=== index? '-3%':'0%',
    
                }
            }
        ))
    )

    

    return(
       <main className="Shop_Main">
        <div className="area_cards">
       
            {springs.map(
                (
                    {boxShadow, top ,left},
                    i
                )=>(
                    
                    <div key={i}   className="card" >  
                        <img src={plantsData[i].picture} className="img_card" alt="plant"></img>
                        <div className="info_part_card">

                            <h2>{plantsData[i].name}</h2>
                            <h4>Price : {` ${plantsData[i].price} $`}</h4>

                            <div className="buttons">
                                <div className="btn_around"

                                    onMouseEnter={()=>{
                                        setIndex(i);
                                        setViewBtn(true)
                                        
                                    }}
                                    onMouseLeave={()=>{
                                        setIndex(null)
                                        setViewBtn(false)
                                       
                                        
                                        
                                        
                                    }}
                                >
                                <animated.button 
                                    className="Card_Btn"
                                    style={!viewBtn? { boxShadow,top,left}:null}
                                    
                                >VIEW</animated.button>
                                </div>
                                <div
                                    className="btn_around"
                                    onMouseEnter={()=>{
                                        setIndex(i);
                                        setBuyBtn(true)

                                       
                                    }}
                                    onMouseLeave={()=>{
                                        setIndex(null)
                                        setBuyBtn(false)
                                       
                                    }}
                                >
                                <animated.button 

                                    className="Card_Btn"
                                    style={BuyBtn ? { boxShadow,top,left}:null}
                                    
                                >BUY</animated.button>
                               </div>
                                
                            </div>
                        </div>
                    </div>
                )
            )}
            
        </div>
       </main>
    )
}

export default ShopPage