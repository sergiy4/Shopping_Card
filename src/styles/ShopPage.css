.Shop_Main{
    width: 100%;
    height: auto;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: rgb(226, 255, 237);
}

.area_cards{
    /* display: flex;
    flex-wrap: wrap; */
   padding: 100px;
    /* background-color: aqua; */
    display: grid;
    grid-template-columns: 1fr 1fr ;
    gap:70px
    /* width: 70%; */
}

.card{
    position: relative;
    height: 550px;
    width: 400px ;
    border: 3px solid black;
    box-sizing:border-box;
    overflow: hidden;
}
.img_card{
    height: 70%;
    width: 100%;
    object-fit: contain;
   
    box-sizing: border-box;
    background-color: #fff;
}
.info_part_card{
    border-top: 3px solid black;
    box-sizing: border-box;
    height: 30%;
    background-color: rgb(37, 255, 164);
    color:rgb(11, 173, 87);
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    gap:8px;
}
.buttons{
    width: 100%;
    display: flex;
    justify-content: space-around;
    
}
.btn_around{
   
    padding-right: 10px;
    padding-bottom: 10px;
    
}
.Card_Btn{
    cursor: pointer;
    position: relative;
    height: 50px;
    width: 100px;
    letter-spacing: 4px;
    border-radius: 0;
}