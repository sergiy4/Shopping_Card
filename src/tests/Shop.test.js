import {render, screen,cleanup, waitFor} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import ShopPage from '../components/ShopPage'
import "@testing-library/jest-dom"

afterEach(()=>{
    cleanup();
})

describe('testing ShopPage',()=>{
    test('correctly render',()=>{
        const {asFragment} = render(<ShopPage/>)

        expect(asFragment()).toMatchSnapshot();
    })

    test('all cards renders', async()=>{
        render(<ShopPage/>)

        const cards = await screen.findAllByRole('img',{name:'plant'})

        waitFor(()=>expect(cards).toBeInTheDocument()) 
        expect(cards.length).toEqual(10)
        
    })
})