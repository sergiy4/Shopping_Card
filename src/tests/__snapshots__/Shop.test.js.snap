// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ShopPage correctly render 1`] = `
<DocumentFragment>
  <main
    class="Shop_Main"
  >
    <div
      class="area_cards"
    >
      <div
        class="card"
      >
        <img
          alt="plant"
          class="img_card"
          src="aglonema.png"
        />
        <div
          class="info_part_card"
        >
          <h2>
            Chinese Evergreen
          </h2>
          <h4>
            Price :  17 $
          </h4>
          <div
            class="buttons"
          >
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
                style="box-shadow: 0px 0px 0px 0px rgb(0, 0, 0); top: 0%; left: 0%;"
              >
                VIEW
              </button>
            </div>
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
              >
                BUY
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <img
          alt="plant"
          class="img_card"
          src="AsparagusFern.jpg"
        />
        <div
          class="info_part_card"
        >
          <h2>
            Asparagus Fern
          </h2>
          <h4>
            Price :  17 $
          </h4>
          <div
            class="buttons"
          >
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
                style="box-shadow: 0px 0px 0px 0px rgb(0, 0, 0); top: 0%; left: 0%;"
              >
                VIEW
              </button>
            </div>
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
              >
                BUY
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <img
          alt="plant"
          class="img_card"
          src="Orchid.png"
        />
        <div
          class="info_part_card"
        >
          <h2>
            Orchid
          </h2>
          <h4>
            Price :  65 $
          </h4>
          <div
            class="buttons"
          >
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
                style="box-shadow: 0px 0px 0px 0px rgb(0, 0, 0); top: 0%; left: 0%;"
              >
                VIEW
              </button>
            </div>
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
              >
                BUY
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <img
          alt="plant"
          class="img_card"
          src="PeaceLily.png"
        />
        <div
          class="info_part_card"
        >
          <h2>
            Peace Lily
          </h2>
          <h4>
            Price :  5 $
          </h4>
          <div
            class="buttons"
          >
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
                style="box-shadow: 0px 0px 0px 0px rgb(0, 0, 0); top: 0%; left: 0%;"
              >
                VIEW
              </button>
            </div>
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
              >
                BUY
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <img
          alt="plant"
          class="img_card"
          src="Philodendron.png"
        />
        <div
          class="info_part_card"
        >
          <h2>
            Philodendron
          </h2>
          <h4>
            Price :  24 $
          </h4>
          <div
            class="buttons"
          >
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
                style="box-shadow: 0px 0px 0px 0px rgb(0, 0, 0); top: 0%; left: 0%;"
              >
                VIEW
              </button>
            </div>
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
              >
                BUY
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <img
          alt="plant"
          class="img_card"
          src="RubberPlant.png"
        />
        <div
          class="info_part_card"
        >
          <h2>
            Rubber Plant
          </h2>
          <h4>
            Price :  25 $
          </h4>
          <div
            class="buttons"
          >
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
                style="box-shadow: 0px 0px 0px 0px rgb(0, 0, 0); top: 0%; left: 0%;"
              >
                VIEW
              </button>
            </div>
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
              >
                BUY
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <img
          alt="plant"
          class="img_card"
          src="SnakePlant.png"
        />
        <div
          class="info_part_card"
        >
          <h2>
            Snake Plant
          </h2>
          <h4>
            Price :  35 $
          </h4>
          <div
            class="buttons"
          >
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
                style="box-shadow: 0px 0px 0px 0px rgb(0, 0, 0); top: 0%; left: 0%;"
              >
                VIEW
              </button>
            </div>
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
              >
                BUY
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <img
          alt="plant"
          class="img_card"
          src="StringofPearls.png"
        />
        <div
          class="info_part_card"
        >
          <h2>
            String of Pearls
          </h2>
          <h4>
            Price :  11 $
          </h4>
          <div
            class="buttons"
          >
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
                style="box-shadow: 0px 0px 0px 0px rgb(0, 0, 0); top: 0%; left: 0%;"
              >
                VIEW
              </button>
            </div>
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
              >
                BUY
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <img
          alt="plant"
          class="img_card"
          src="Yucca.png"
        />
        <div
          class="info_part_card"
        >
          <h2>
            Yucca
          </h2>
          <h4>
            Price :  44 $
          </h4>
          <div
            class="buttons"
          >
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
                style="box-shadow: 0px 0px 0px 0px rgb(0, 0, 0); top: 0%; left: 0%;"
              >
                VIEW
              </button>
            </div>
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
              >
                BUY
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <img
          alt="plant"
          class="img_card"
          src="ZZPlant.png"
        />
        <div
          class="info_part_card"
        >
          <h2>
            ZZ Plant
          </h2>
          <h4>
            Price :  48 $
          </h4>
          <div
            class="buttons"
          >
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
                style="box-shadow: 0px 0px 0px 0px rgb(0, 0, 0); top: 0%; left: 0%;"
              >
                VIEW
              </button>
            </div>
            <div
              class="btn_around"
            >
              <button
                class="Card_Btn"
              >
                BUY
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
